migration/migration.go

```diff
@@ -238,7 +239,10 @@ func uploadCharms(config UploadBinariesConfig) error {
 	// It is critical that charms are uploaded in ascending charm URL
 	// order so that charm revisions end up the same in the target as
 	// they were in the source.
-	naturalsort.Sort(config.Charms)
+	_, err := naturalsort.Sort(config.Charms)
```

Comment:
For this use case, ie charm revisions, we never expect an error, but if we get one, as per the code comment above, we can't just fall back to a generic sort